EchoPro/survey.py

```diff
@@ -327,269 +328,502 @@ def read_validated_data( self ,
+        """
+        Imputes sigma_bs for strata without measurements or values
+
+        Parameters
+        ----------
+        species_id : np.float64
+            Numeric code representing a particular species of interest
+
+        Notes
+        -----
+        This function iterates through all stratum layers to impute either the
+        nearest neighbor or mean sigma_bs for strata that are missing values.
+        """    
+        #### TODO: CURRENTLY : species_id is unused since only hake are being processed, but this will need
+        ### to actually be parameterized in the future
+        # Collect all possible strata values
+        strata_options = np.unique(self.spatial['strata_df'].copy().stratum_num)
+        
+        #
+        strata_mean = self.acoustics['sigma_bs']['strata_mean'].copy()
+        
+        # impute missing strata values
+        present_strata = np.unique(strata_mean['stratum_num']).astype(int)
+        missing_strata = strata_options[~(np.isin(strata_options, present_strata))]
+        
+        if len(missing_strata) > 0:            
+            sigma_bs_impute = pd.concat([strata_mean , 
+                                            pd.DataFrame({'stratum_num': missing_strata ,
+                                                         'species_id': np.unique( strata_mean.species_id ) ,
+                                                         'mean_sigma_bs': np.nan})]).sort_values('stratum_num')        
+            
+            # Find strata intervals to impute over        
+            for i in missing_strata:
+                strata_floor = present_strata[present_strata < i]
+                strata_ceil = present_strata[present_strata > i]
+
+                new_stratum_below = np.max(strata_floor) if strata_floor.size > 0 else None
+                new_stratum_above = np.min(strata_ceil) if strata_ceil.size > 0 else None      
+                
+                sigma_bs_indexed = sigma_bs_impute[sigma_bs_impute['stratum_num'].isin([new_stratum_below, new_stratum_above])]
+                
+                sigma_bs_impute.loc[sigma_bs_impute.stratum_num==i , 'mean_sigma_bs'] = sigma_bs_indexed['mean_sigma_bs'].mean()
+                
+            self.acoustics['sigma_bs']['strata_mean'] = sigma_bs_impute        
+   
+    
+    def fit_binned_length_weight_relationship( self ,
+                                               species_id: np.float64 ):
+        """
+        Fit a length-weight relationship across discrete bins
+
+        Parameters
+        ----------
+        species_id : np.float64
+            Numeric code representing a particular species of interest
+
+        Notes
+        -----
+        This function first fits a length-weight regression based on measured 
+        values and then produces an array of fitted weight values (based on 
+        binned length values) that are used for later biomass calculations, 
+        weighting, and apportionment.
```

Comment:
Minor edits to improve clarity.
```suggestion
        Notes
        -----
        This function first fits a length-weight regression based on measured 
        values and then produces an array of fitted weight values based on 
        binned length values.
        The length-weight relationship produced here are used for later 
        biomass calculations and apportionment.
```